src/provider/provider_os_memory_posix.c

```diff
@@ -106,3 +109,39 @@ void os_strerror(int errnum, char *buf, size_t buflen) {
         LOG_PERR("Retrieving error code description failed");
     }
 }
+
+// open a devdax
+int os_devdax_open(const char *path) {
+    if (path == NULL) {
+        LOG_ERR("empty path");
+        return -1;
+    }
+
+    if (strstr(path, "/dev/dax") != path) {
+        LOG_ERR("path of the file \"%s\" does not start with \"/dev/dax\"",
+                path);
+        return -1;
+    }
+
+    int fd = open(path, O_RDWR);
+    if (fd == -1) {
+        LOG_PERR("cannot open the file: %s", path);
+        return -1;
+    }
+
+    struct stat statbuf;
+    int ret = stat(path, &statbuf);
+    if (ret) {
+        LOG_PERR("stat(%s) failed", path);
+        close(fd);
+        return -1;
+    }
+
+    if (!S_ISCHR(statbuf.st_mode)) {
+        LOG_ERR("file %s is not a character device", path);
```

Comment:
we could add negative tests for all these error paths... (wrong device, no path, ...) for better coverage, I guess

Reply:
Done